lib/hammer_cli_katello/content_view_purge.rb

```diff
@@ -74,32 +68,20 @@ def execute
 
     private def purge_version(v)
       if option_async?
-        task = resource_content_view_versions.call(:destroy, 'id' => v["id"])
+        task = call(:destroy, :content_view_versions, 'id' => v["id"])
         print_message _("Version '%{version}' of content view '%{view}' scheduled "\
                         "for deletion in task '%{task_id}'.") %
-                      {version: v["version"], view: content_view["name"], task_id: task['id']}
+                      {version: v["version"], view: v["content_view"]["name"], task_id: task['id']}
       else
-        task_progress(resource_content_view_versions.call(:destroy, 'id' => v["id"]))
+        task_progress(call(:destroy, :content_view_versions, 'id' => v["id"]))
         print_message _("Version '%{version}' of content view '%{view}' deleted.") %
-                      {version: v["version"], view: content_view["name"]}
+                      {version: v["version"], view: v["content_view"]["name"]}
       end
     end
 
-    private def content_view
-      @content_view ||= resource_content_views.call(:show, 'id' => options['option_id'])
-    end
-
-    private def old_unused_versions
-      return @old_unused_versions if @old_unused_versions
-
-      all_versions = content_view['versions'].sort_by do |version|
-        [version[:major], version[:minor]]
-      end
```

Comment:
We need to delete the oldest versions. We can either have the API return the versions in order by version or keep this sort. Right now if you purge a CV, it removes the most recent versions, which is not what we want.

Reply:
Good call, forgot to consider that! I've updated with the appropriate sorting, and added destroy method in apipie_helper and updated tests as well. Thanks!